doc/whats_new.rst

```diff
@@ -278,7 +278,11 @@ Bug fixes
       ``solver='liblinear'`` did not accept ``class_weights='balanced``.
       (`#6817 <https://github.com/scikit-learn/scikit-learn/pull/6817>`_).
       By `Tom Dupre la Tour`_.
-
+      
+    - Fix bug in :class: `neighbors.RadiusNeighborsClassifier` where the presence of 
+      labelled outliers was creating a bug when trying to concatenate predicted labels 
```

Comment:
I wouldn't worry the user about the mechanism of the logic error, but to make clear that the error occurred when there were outliers being labelled and a weight function.